src/libs/actions/Report.js

```diff
@@ -164,21 +167,15 @@ function fetchChatReportsByIDs(chatList) {
             // variable called simplifiedReports which hold the participants (minus the current user) for each report.
             // Using this simplifiedReport we can call PersonalDetails.getFromReportParticipants to get the
             // personal details of all the participants and even link up their avatars to report icons.
-            const simplifiedReports = [];
+            const simplifiedReports = {};
             _.each(fetchedReports, (report) => {
-                const newReport = getSimplifiedReportObject(report);
-                simplifiedReports.push(newReport);
-
-                if (lodashGet(report, 'reportNameValuePairs.type') === 'chat') {
-                    newReport.reportName = getChatReportName(report.sharedReportList);
-                }
-
-                // Merge the data into Onyx
-                Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`, newReport);
+                const key = `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`;
+                simplifiedReports[key] = getSimplifiedReportObject(report);
             });
+            Onyx.mergeCollection(ONYXKEYS.COLLECTION.REPORT, simplifiedReports);
```

Comment:
Might be a good opportunity to briefly explain why `mergeCollection` is used here.